packages/flutter/test/cupertino/refresh_test.dart

```diff
@@ -1114,6 +1114,58 @@ void main() {
       },
     );
 
+    testWidgets(
+      'builder not called when lastIndicatorExtent is less than 0',
+      (WidgetTester tester) async {
+        debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+        refreshIndicator = const Center(child: Text('-1'));
+
+        await tester.pumpWidget(
+          Directionality(
+            textDirection: TextDirection.ltr,
+            child: CustomScrollView(
+              slivers: <Widget>[
+                CupertinoSliverRefreshControl(
+                  builder: builder,
+                  onRefresh: onRefresh,
+                ),
+                buildAListOfStuff(),
+              ],
+            ),
+          ),
+        );
+
+        // Ensure that in inactive state builder was not called
+        expect(
+          find.widgetWithText(Center, '-1'),
+          findsNothing,
+        );
+
+        // Scrolling down to ensure builder was called and refreshIndicator
+        // is present
+        final TestGesture gesture = await tester.startGesture(const Offset(0.0, 0.0));
+        await gesture.moveBy(const Offset(0.0, 150.0));
+        await tester.pump();
+
+        expect(
```

Comment:
We're not really exercising the new logic in this test. The builder is always called as soon as you start dragging down. The fix was to make sure it's still called when it snapped back more than 90% of the way. 

I'd look at the test 'expanded refreshing sliver goes away when done'. Make sure it's pulled in far enough to actually trigger the refresh. Then let it retract until it's 90% of the way back. Then make sure the builder is still called.

Reply:
Would it be enough to scroll to 100 * 0.05 (saying that previous scrolling offset was Offset(0, 100)) and check that Center(child: Text(‘-1’)) returned from builder is still present?